src/communication/network_interface_sr.c

```diff
@@ -10698,3 +10698,231 @@ scdc_end_session (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int
 
   return;
 }
+
+static int
+flashback_verify_time (THREAD_ENTRY * thread_p, time_t start_time, time_t end_time, LOG_LSA * start_lsa,
+		       LOG_LSA * end_lsa)
+{
+  int error_code = NO_ERROR;
+  time_t ret_time = 0;
+
+
+  /* 1. Check start_time */
+  if (start_time < log_Gl.hdr.db_creation)
+    {
+      return ER_FLASHBACK_INVALID_TIME;
+    }
+  else
+    {
+      ret_time = start_time;
+
+      error_code = cdc_find_lsa (thread_p, &ret_time, start_lsa);
+      if (error_code == NO_ERROR || error_code == ER_CDC_ADJUSTED_LSA)
+	{
+	  /* find log record at the time
+	   * ret_time : time of the commit record or dummy record that is found to be greater than or equal to the start_time at first */
+
+	  if (ret_time > end_time)
```

Comment:
If they are the same, the only log record we can get is a donetime record. and it is meaningless. Isn't it?

Reply:
+ I thought `start_time`- 1 sec is used here.

Reply:
yes, it is meaningless. 

`start_time (returned from cdc_find_lsa) -1 ` was decided to use when making summary info per transaction, when fill the start_time of the transaction. 
 
but it is better to do here too, and maybe client-side validation (start_time >= end_time) will be modified to (start_time > end_time).

Reply:
sorry for above comment. please ignore it. 
1. (start_time <  end_time) is right  : start_time never be same with end_time. 
2. returned time from cdc_find_lsa() can not be same with end_time. 
    -  `if (ret_time >= end_time) return error; `

Reply:
3. cdc_find_lsa() returns `start_lsa`
  - compared with log record containing time info : LOG_COMMIT, LOG_ABORT, or LOG_DUMMY_HA_SERVER_STATE  
  - traversing logs in time order and if input time >= time of log record, then return time of log record, and LSA. 

So, I think `start_time -1` is not required here, because it can return LSA of former log record.